src/ResourceManager/Resources/Commands.Resources.UnitTests/RoleAssignmentUnitTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Commands.Resources.Models.Authorization;
+using Microsoft.Azure.Commands.Resources;
+using System.Collections.Generic;
+
+namespace Commands.Resources.UnitTests
+{
+    [TestClass]
+    public class RoleAssignmentUnitTests
+    {
+
+        private void VerifyInvalidScope(string scope, string error)
+        {
+            try
+            {
+                AuthorizationClient.ValidateScope(scope);
+                Assert.Fail();
+            }
+            catch(ArgumentException ex)
+            {
+                Assert.AreEqual(ex.Message, error);
+            }
+        }
+
+        [TestMethod]
+        public void VerifyInvalidScopes()
+        {
+            Dictionary<string, string> scopeAndErrors = new Dictionary<string, string>();
+            scopeAndErrors.Add("/subscriptions/e9ee799d-6ab2-4084-b952-e7c86344bbab/Should be 'ResourceGroups'/any group name", "Scope '/subscriptions/e9ee799d-6ab2-4084-b952-e7c86344bbab/Should be 'ResourceGroups'/any group name' should begin with '/subscriptions/<subid>/resourceGroups'.");
```

Comment:
No.  The entire idea is to cover a much larger range of inputs and outputs, which is much easier in a unit test.

Please cover positive and negative cases, null and empty values, and other important input equivalence classes.